pkg/entrypoint/admissioncontroller/admissioncontroller.go

```diff
@@ -0,0 +1,217 @@
+package admissioncontroller
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	"github.com/openshift/origin/pkg/security/apis/security"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kjson "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+
+	"github.com/openshift/openshift-azure/pkg/util/log"
+)
+
+var (
+	serializer = kjson.NewSerializer(kjson.DefaultMetaFactory, legacyscheme.Scheme, legacyscheme.Scheme, false)
+	codec      = legacyscheme.Codecs.CodecForVersions(nil, serializer, nil, runtime.InternalGroupVersioner)
+)
+
+func toInt32Ptr(i int32) *int32 {
+	return &i
+}
+
+func toBoolPtr(b bool) *bool {
+	return &b
+}
+
+type admissionControllerConfig struct {
+	Whitelist []string `json:"whitelist"`
+}
+
+func (c *admissionControllerConfig) loadConfig(log *logrus.Entry) *admissionControllerConfig {
+	configFile, err := ioutil.ReadFile("/etc/aro-admission-controller/aro-admission-controller.yaml")
+	if err != nil {
+		log.Printf("Error reading config file %s", err)
+	}
+	err = yaml.Unmarshal(configFile, c)
+	if err != nil {
+		log.Fatalf("Error unmarshalling config file %s", err)
+	}
+	err = c.validate()
+	if err != nil {
+		log.Fatalf("Error interpreting config file %s", err)
+	}
+	return c
+}
+
+func (c *admissionControllerConfig) validate() error {
+	var err error
+	for _, w := range c.Whitelist {
+		_, err = regexp.Compile(w)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (ac *admissionController) handleHealthz(w http.ResponseWriter, r *http.Request) {
+	return
+}
+
+type admissionController struct {
+	log               *logrus.Entry
+	client            internalclientset.Interface
+	restricted        *security.SecurityContextConstraints
+	whitelistedImages []*regexp.Regexp
+	protectedSCCs     map[string]security.SecurityContextConstraints
+}
+
+func (ac *admissionController) getAdmissionReviewRequest(r *http.Request) (req *admissionv1beta1.AdmissionRequest, errorcode int) {
+	ac.log.Printf("New review request %s", r.RequestURI)
+	if r.Method != http.MethodPost {
+		return nil, http.StatusMethodNotAllowed
+	}
+	if r.Header.Get("Content-Type") != "application/json" {
+		return nil, http.StatusUnsupportedMediaType
+	}
+
+	var reviewIncoming *admissionv1beta1.AdmissionReview
+	err := json.NewDecoder(r.Body).Decode(&reviewIncoming)
+	if err != nil {
+		return nil, http.StatusBadRequest
+	}
+	req = reviewIncoming.Request
```

Comment:
I still need some data from the review request, and it's different data (operation in SCC validation, requesting user in Pod validation). This would either make the getAdmissionReviewRequest function code path specific or return unnecessary values.

Reply:
Explained on call.
Done.